Small type error in terserOptions

after upgrading to the latest 5.2.2 (thanks for that other fix) i do see another small issue

```
error TS2345: Argument of type '{ parallel: true; terserOptions: { compress: { dead_code: boolean; 
pure_funcs: string[]; }; mangle: boolean; }; }' is not assignable to parameter of type
 'BasePluginOptions & { minify: MinimizerImplementation<unknown>; terserOptions?: unknown; }'.
  Property 'minify' is missing in type '{ parallel: true; terserOptions: { compress: 
{ dead_code: boolean; pure_funcs: string[]; }; mangle: boolean; }; }' but required in
 type '{ minify: MinimizerImplementation<unknown>; terserOptions?: unknown; }'.
```

```
(alias) new TerserPlugin<unknown>(options?: (BasePluginOptions & {
    minify: MinimizerImplementation<unknown>;
    terserOptions?: unknown;
}) | undefined): TerserPlugin<unknown>
import TerserPlugin
```

![image](https://user-images.githubusercontent.com/3482445/131980621-910cc826-5eaa-4410-97ab-1a99dbe10d3d.png)


I think this is caused by the second minify not being optional (?)

```
export type DefinedDefaultMinimizerAndOptions<T> =
  InferDefaultType<T> extends TerserOptions
    ? {
        minify?: MinimizerImplementation<InferDefaultType<T>> | undefined;
        terserOptions?: InferDefaultType<T> | undefined;
      }
    : {
        minify: MinimizerImplementation<InferDefaultType<T>>;
        terserOptions?: InferDefaultType<T> | undefined;
      };
```

thanks

The related TS issue is that the `terserMinify` imported from "terser-webpack-plugin" doesn't get accepted by `TerserPlugin` constructor when passed as `minify` option (that `terserMinify` named as used by default):

```
error TS2322: Type 'typeof terserMinify' is not assignable to type 'MinimizerImplementation<unknown>'.
  Type 'typeof terserMinify' is not assignable to type 'BasicMinimizerImplementation<unknown>'.
    Types of parameters 'minimizerOptions' and 'minifyOptions' are incompatible.
      Type 'unknown' is not assignable to type 'PredefinedOptions & CustomOptions'.
        Type 'unknown' is not assignable to type 'PredefinedOptions'.

86                         minify: terserMinify
```

--- a/package.json
+++ b/package.json
@@ -17,8 +17,7 @@
     "node": ">= 10.13.0"
   },
   "scripts": {
-    "start": "npm run build -- -w",
-    "clean": "del-cli dist",
+    "clean": "del-cli dist types",
     "prebuild": "npm run clean",
     "build:types": "tsc --declaration --emitDeclarationOnly --outDir types && prettier \"types/**/*.ts\" --write && prettier types --write",
     "build:code": "cross-env NODE_ENV=production babel src -d dist --copy-files",

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -207,7 +207,7 @@ export type DefinedDefaultMinimizerAndOptions<T> =
 /**
  * @template T
  */
-declare class TerserPlugin<T = TerserOptions> {
+declare class TerserPlugin<T> {
   /**
    * @private
    * @param {any} input

